src/topology/config/mod.rs

```diff
@@ -331,6 +361,28 @@ impl Config {
         self.global.dns_servers.sort();
         self.global.dns_servers.dedup();
 
+        let default_log_schema = LogSchemaOptions::default();
+        if with.global.log_schema != default_log_schema {
+            if self.global.log_schema.host_key != default_log_schema.host_key &&
+                self.global.log_schema.host_key != with.global.log_schema.host_key {
+                errors.push("conflicting values for 'log_schema.host_key' found".to_owned());
+            } else {
+                self.global.log_schema.host_key = with.global.log_schema.host_key;
+            }
+            if self.global.log_schema.message_key != default_log_schema.message_key &&
+                self.global.log_schema.message_key != with.global.log_schema.message_key {
+                errors.push("conflicting values for 'log_schema.message_key' found".to_owned());
+            } else {
+                self.global.log_schema.message_key = with.global.log_schema.message_key;
+            }
+            if self.global.log_schema.timestamp_key != default_log_schema.timestamp_key &&
+                self.global.log_schema.timestamp_key != with.global.log_schema.timestamp_key {
+                errors.push("conflicting values for 'log_schema.timestamp_key' found".to_owned());
+            } else {
+                self.global.log_schema.timestamp_key = with.global.log_schema.timestamp_key;
+            }
+        }
```

Comment:
I'm unclear as to what the conflicting values could come from here. Maybe it would be worth adding a comment to help my ignorance?